src/persist/src/operators/source.rs

```diff
@@ -76,14 +71,14 @@ fn listen_source<G, K, V>(
     // initial frontier because an empty frontier would signal that we are at "the end of time",
     // meaning that there will be no more events in the future.
     initial_frontier: Option<Antichain<u64>>,
-    listen_rx: Receiver<ListenEvent>,
+    mut listen_rx: Option<crossbeam_channel::Receiver<ListenEvent>>,
```

Comment:
why does this take an option now? that seems wrong to me

Reply:
fixed by passing in a receiver directly as an argument to listen